calibrators/docs/tutorials/tag_based_pnp_calibrator.md

```diff
@@ -0,0 +1,157 @@
+
+When the user starts the calibration, `rviz` and the `image view` should be displayed like below.
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/visualization1.jpg"  alt="visualization1">
+</p>
+
+After the tools detect the lidartag and apriltag, it will show the detection markers on the `rviz` and the `image view`. The text in rviz will also display the number of pairs of lidartag/apriltag detections.
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/visualization2.jpg"  alt="visualization2">
+</p>
+
+Once the user observes the number of pairs increases in the `rviz`, he can start moving the tag to another position.
+
+While moving the tag to another position, the user should move the tag decisively from one place to the other, not stopping and starting mid-way.
+Please make sure the new position is at least `calibration_min_pair_distance` from the closest detection in the set of converged detections and the tag is in the FOV of both lidar and camera.
+
+Depending on the mount and the ground surface, the tags can oscillate quite a bit, which can be detrimental to the calibration process. In this case, the user can stop the tag with its hand.
+
+At the end of the calibration, we can get 8 detection pairs which are shown below.
+
+![tag_based_pnp_calibrator](../images/tag_based_pnp_calibrator/visualization3.jpg)
+
+The output in the console should be as follows:
+
+```text
+[tag_based_pnp_calibrator_node]:  Initial reprojection error=15.41
+[tag_based_pnp_calibrator_node]:  Current reprojection error=1.09
+[tag_based_pnp_calibrator_node]:  Filtered reprojection error=1.10
+```
+
+## Results
+
+After the calibration process finishes, the `sensor_calibration_manager` will display the results in the UI and allow the user to save the calibration data to a file.
+
+In the UI of the X2 project, three different TF trees are displayed: `Initial TF Tree`, `Calibration Tree`, and `Final TF Tree`.
+
+- The `Initial TF Tree` presents the initial TF connections between sensors needed for calibration.
+- The `Calibration Tree` shows the calibrated transformation between sensors, in this tutorial, `camera6/camera_optical_link` to `pandar_40p_front`.
+- The `Final TF Tree` depicts the TF tree after incorporating the updated calibrated transformation. As the Autoware project utilizes the concept of [sensor_kit](https://autowarefoundation.github.io/autoware-documentation/main/how-to-guides/integrating-autoware/creating-vehicle-and-sensor-model/creating-sensor-model/), the final transformation we need is `front_unit_base_link` to `camera6/camera_link`. We calculate the final transformation in [sensor_calibration_manager](https://github.com/tier4/CalibrationTools/blob/tier4/universe/sensor/sensor_calibration_manager/sensor_calibration_manager/calibrators/x2/tag_based_pnp_calibrator.py). Since the final transformation changed after the calculations, the arrow is shown in red.
```

Comment:
```suggestion
- The `Final TF Tree` depicts the TF tree after incorporating the updated calibrated transformation. As autoware utilizes the concept of [sensor_kit](https://autowarefoundation.github.io/autoware-documentation/main/how-to-guides/integrating-autoware/creating-vehicle-and-sensor-model/creating-sensor-model/), the final transformation required to comply to the specifications is `front_unit_base_link` to `camera6/camera_link`. This transformation in performed by the [sensor_calibration_manager](https://github.com/tier4/CalibrationTools/blob/tier4/universe/sensor/sensor_calibration_manager/sensor_calibration_manager/calibrators/x2/tag_based_pnp_calibrator.py). The red arrow indicates that the final transformation changed after the calibration process.
```

Reply:
@vividf 
In addition, it is not the sensor_calibration_manager. It is the calibrator interface related to this project (link here).
Links inside this repository should be relative not absolute

Reply:
fixed in https://github.com/tier4/CalibrationTools/pull/164/commits/f47f969364d87c4b717003361029d58d09e75aaa